src/test/java/com/meilisearch/integration/SettingsTest.java

```diff
@@ -844,6 +846,55 @@ public void testUpdateDistinctAttributeSettingsUsingNull() throws Exception {
         assertEquals(initialDistinctAttribute, resetDistinctAttribute);
     }
 
+    /** Tests of the pagination setting methods */
+    @Test
+    @DisplayName("Test get pagination settings by uid")
+    public void testGetPaginationSettings() throws Exception {
+        Index index = createIndex("testGetPaginationSettings");
+        Settings initialSettings = index.getSettings();
+        Pagination initialPagination = index.getPaginationSettings();
+
+        assertEquals(initialSettings.getPagination().getMaxTotalHits(),0);
+        assertNotNull(initialPagination.getMaxTotalHits());
+    }
+    @Test
+    @DisplayName("Test update pagination settings")
+    public void testUpdatePaginationSettings() throws Exception {
+        Index index = createIndex("testUpdatePaginationSettings");
+        Pagination newPagination = new Pagination();
+
+        Integer MaxTotalHitsTypos = 100;
+
+        newPagination.setMaxTotalHits(MaxTotalHitsTypos);
+        index.waitForTask(index.updatePaginationSettings(newPagination).getTaskUid());
+        Pagination updatedPagination = index.getPaginationSettings();
+
+        assertTrue(
+            updatedPagination.getMaxTotalHits() == 100
+        );
+    }
+    @Test
+    @DisplayName("Test reset pagination settings")
+    public void testResetPaginationSettings() throws Exception {
+        Index index = createIndex("testResetPaginationSettings");
+
+        Pagination initialPagination = index.getPaginationSettings();
+        Pagination newPagination = new Pagination();
+        Integer MaxTotalHitsTypos = 100;
+        newPagination.setMaxTotalHits(MaxTotalHitsTypos);
+
+        index.waitForTask(index.resetPaginationSettings().getTaskUid());
+        Pagination paginationAfterReset = index.getPaginationSettings();
+
+        assertTrue(
+            paginationAfterReset.getMaxTotalHits() == 0
```

Comment:
```suggestion
            paginationAfterReset.getMaxTotalHits() == 1000
```
Same as previously the default value of maxTotalHits is 1000